### PR TITLE
Make cli_utils.ArgParser compatible with new argparse.ArgumentParser

### LINKED ISSUE #5724: xonsh exception with Python 3.12.7
## Current Behavior

```
@ xonsh --version
xonsh/0.18.4

♅ 15:42:06 rrs@xps ~/NoBac/Commun/xonsh   master↑·4|✓
@ xonsh --no-rc
rrs@xps ~/NoBac/Commun/xonsh master @ xontrib list
['list']
Exception in {'cls': 'ProcProxy', 'name': None, 'func': <xonsh.xontribs.XontribAlias object at 0x7f4d9e7101a0>, 'alias': 'xontrib', 'pid': 1954810}
xonsh: To log full traceback to a file set: $XONSH_TRACEBACK_LOGFILE = <filename>
Traceback (most recent call last):
  File "/home/rrs/.local/share/pipx/venvs/xonsh/lib/python3.12/site-packages/xonsh/procs/proxies.py", line 784, in wait
    r = run_with_partial_args(
        ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/rrs/.local/share/pipx/venvs/xonsh/lib/python3.12/site-packages/xonsh/cli_utils.py", line 381, in run_with_partial_args
    return func(**kwargs)
           ^^^^^^^^^^^^^^
  File "/home/rrs/.local/share/pipx/venvs/xonsh/lib/python3.12/site-packages/xonsh/cli_utils.py", line 647, in __call__
    result = dispatch(
             ^^^^^^^^^
  File "/home/rrs/.local/share/pipx/venvs/xonsh/lib/python3.12/site-packages/xonsh/cli_utils.py", line 406, in dispatch
    parsed = parser.parse_args(args)
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/argparse.py", line 1918, in parse_args
    args, argv = self.parse_known_args(args, namespace)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/argparse.py", line 1928, in parse_known_args
    return self._parse_known_args2(args, namespace, intermixed=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/argparse.py", line 1957, in _parse_known_args2
    namespace, args = self._parse_known_args(args, namespace, intermixed)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: ArgParser._parse_known_args() takes 3 positional arguments but 4 were given
```
## Expected Behavior
xonsh should spawn without any errors

## xonfig

```
rrs@xps ~/NoBackup/Community/xonsh master [1] @ xonfig
xonsh: For full traceback set: $XONSH_SHOW_TRACEBACK = True
Exception in {'cls': 'ProcProxy', 'name': None, 'func': <xonsh.xonfig.XonfigAlias object at 0x7f6c4eb61040>, 'alias': 'xonfig', 'pid': 1956443}
TypeError: ArgParser._parse_known_args() takes 3 positional arguments but 4 were given
```

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### PR DESCRIPTION
**Pull request by JamesParrott**
tl;dr
Xonsh's code in cli_utils overrides 'private' methods of argparser.ArgumentParser. Three weeks ago these 'private' methods were changed in the core library by the CPython team. Those changes were rolled out in Python 3.12.7

Confirmed fixes #5724 ~~This should hopefully fix #5724 . I'll test this later today~~

I think I've figured it (#5724) out. [This commit](https://github.com/python/cpython/pull/125839/files) in CPython 3 weeks ago changed the signature of the internal method ArgParser._parse_known_args (adding a boolean arg, intermixed). Other methods of the core library's ArgumentParser (in particular _parse_known_args2) call self._parse_known_args with this new signature. In Xonsh, cli_utils.ArgParser, a sub class of ap.ArgumentParser, had previously overridden _parse_known_args.

The old signature of cli_utils.ArgParser._parse_known_args (without intermixed) is incompatible with calls to the new signature of argparse.ArgumentParser._parse_known_args (intermixed doesn't have a default on the 'private' methods, but is set in calls from the 'public' API methods ArgumentParser.parse_known_intermixed_args and ArgumentParser.parse_known_args).

Hopefully the signature of cli_utils.ArgParser._parse_known_args just needs to have intermixed, or *args, **kwargs added to it. The latter would actually be the same fix as @frostming 's commit 3266 for pdm (quoted above).

I still haven't the faintest idea why it only affects gcc builds on Debian Trixie though (it is a testing repo, perhaps that just happens to be where the bug was first discovered). 

Copied from here:  https://github.com/xonsh/xonsh/issues/5724#issuecomment-2466716810

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### CODE REVIEW

**Comment by JamesParrott** [comment]
Confirmed fixes #5724 

```
docker run --rm -it debian:trixie bash
apt-get update -y
apt-get install -y python3.12-venv git
python3.12 -m venv venv
. venv/bin/activate
git clone --depth=1 --branch=patch-2 https://github.com/JamesParrott/xonsh
cd xonsh
pip install -e .[full]
```


```
(venv) root@2e256fcc7a59:/xonsh# xonsh

                                           Welcome to the xonsh shell 0.18.4

                                               ~ Exofrills in the shell ~

------------------------------------------------------------------------------------------------------------------------

                     Create ~/.xonshrc file manually or use xonfig to suppress the welcome message

Start from commands:
  xonfig web         # Run the configuration tool in the browser to create ~/.xonshrc
  xonfig tutorial    # Open the xonsh tutorial in the browser

------------------------------------------------------------------------------------------------------------------------

(venv) root@2e256fcc7a59 /xonsh patch-2 @# xonfig
+-----------------------------+-----------------+
| xonsh                       | 0.18.4          |
| Python                      | 3.12.7          |
| PLY                         | 3.11            |
| have readline               | True            |
| prompt toolkit              | 3.0.48          |
| shell type                  | prompt_toolkit  |
| history backend             | json            |
| pygments                    | 2.18.0          |
| on posix                    | True            |
| on linux                    | True            |
| distro                      | debian          |
| on wsl                      | False           |
| on darwin                   | False           |
| on windows                  | False           |
| on cygwin                   | False           |
| on msys2                    | False           |
| is superuser                | True            |
| default encoding            | utf-8           |
| xonsh encoding              | utf-8           |
| encoding errors             | surrogateescape |
| xontrib                     | []              |
| RC file                     | []              |
| UPDATE_OS_ENVIRON           | False           |
| XONSH_CAPTURE_ALWAYS        | False           |
| XONSH_SUBPROC_OUTPUT_FORMAT | stream_lines    |
| THREAD_SUBPROCS             | True            |
| XONSH_CACHE_SCRIPTS         | True            |
+-----------------------------+-----------------+
(venv) root@2e256fcc7a59 /xonsh patch-2 @# xontrib
coreutils  not-loaded
```

**Comment by gforsyth** [comment]
Hey @JamesParrott -- thanks for putting this in! I think this was also fixed in #5722 so I'm going to close this out, but if the error continues showing up please ping and we can reopen this.

**Comment by JamesParrott** [comment]
@gforsyth Thanks for looking this over.  Unfortunately both 
a) [the bug](https://github.com/xonsh/xonsh/blob/f2e4c93785c3c8490ccc817f758ac9d3f72e73b3/xonsh/cli_utils.py#L357) is still in Xonsh's code - the override of _parse_known_args is still using the old signature, and 
b) the error (#5724) still occurs even with the latest commit (see below), 

So yes please, could you re-open this PR and seriously consider it for inclusion?  The proposed two line change mimics a similar change used in the PDM project, and just add *args, and **kwargs and pass them through to super calls, so are neither demanding to understand, nor controversial.

I know it's a tricky bug to reproduce, hence why I provided detailed steps to do so.  But I'm confident this trickiness is simply because the Debian Trixie team discovered it first,a nd other distros and users haven't caught up yet.  The bug also occurs in a Python built from source too, and is solely due to pure Python code in the stdlib and xonsh.

```
root@docker-ce-ubuntu-4gb-nbg1-2:~# docker run --rm -it debian:trixie bash
root@6214f2eac1d8:/# apt-get update -y
root@6214f2eac1d8:/# apt-get install -y python3.12-venv
root@6214f2eac1d8:/# python3.12 -m venv venv
root@6214f2eac1d8:/# . venv/bin/activate
(venv) root@6214f2eac1d8:/# git
bash: git: command not found
(venv) root@6214f2eac1d8:/# apt-get install -y git
(venv) root@6214f2eac1d8:/# pip install git+https://github.com/xonsh/xonsh.git@f2e4c93785c3c8490ccc817f758ac9d3f72e73b3
(venv) root@6214f2eac1d8:/# xonsh

                                         Welcome to the xonsh shell 0.18.4.dev3

                              ~ Piggy glanced nervously into hell and cradled the xonsh ~

------------------------------------------------------------------------------------------------------------------------

                     Create ~/.xonshrc file manually or use xonfig to suppress the welcome message

Start from commands:
  xonfig web         # Run the configuration tool in the browser to create ~/.xonshrc
  xonfig tutorial    # Open the xonsh tutorial in the browser

You are currently using the readline backend. For interactive tab-completion, on-the-fly syntax highlighting, and more, install prompt_toolkit by running:

  xpip install -U 'xonsh[full]'

------------------------------------------------------------------------------------------------------------------------

(venv) root@6214f2eac1d8 / @# xonfig
xonsh: For full traceback set: $XONSH_SHOW_TRACEBACK = True
Exception in {'cls': 'ProcProxy', 'name': None, 'func': <xonsh.xonfig.XonfigAlias object at 0x71ce7b47dfa0>, 'alias': 'xonfig', 'pid': 3570}
TypeError: ArgParser._parse_known_args() takes 3 positional arguments but 4 were given
```

**Comment by JamesParrott** [comment]
I've raised another PR (#5732) which duplicates this one, so that the restored CI checks are run.  This also avoids the need to rebase.
